pandas/_libs/lib.pyx

```diff
@@ -2208,31 +2208,47 @@ def maybe_convert_objects(ndarray[object] objects, bint try_float=0,
     return objects
 
 
+_no_default = object()
+
+
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def map_infer_mask(ndarray arr, object f, const uint8_t[:] mask, bint convert=1):
+def map_infer_mask(ndarray arr, object f, const uint8_t[:] mask, bint convert=1,
+                   object na_value=_no_default, object dtype=object):
     """
     Substitute for np.vectorize with pandas-friendly dtype inference
 
     Parameters
     ----------
     arr : ndarray
     f : function
+    mask : ndarray
```

Comment:
`ndarray[uint8]`

Reply:
How do we handle dtypes in docstrings? I didn't think numpy supported that yet.

Reply:
this is the pattern we use in the cython docstrings, but we're not super-consistent about it.  looking forward to if/when numpy supports that kind of annotation